solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -615,6 +614,22 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
         // verify that the protected liquidity is not removed on the same block in which it was added
         require(liquidity.timestamp < time(), "ERR_TOO_EARLY");
 
+        // get the various rates between the reserves upon adding liquidity and now
+        PackedRates memory packedRates =
+            packRates(
+                liquidity.poolToken,
+                liquidity.reserveToken,
+                liquidity.reserveRateN,
+                liquidity.reserveRateD
+            );
+
+        verifyRate(
```

Comment:
cleaner implementation :) I'd just add a comment both here and in addLiquidity that explains why we verify the deviation so early

Reply:
done